Add support for ggalt::geom_dumbbell.

trekonom opened the pull request:
This PR adds support for `ggalt::geom_dumbbell`  and closes #1454 . 

- The implementation follows `ggalt::GeomDumbbell`.

- Tooltips only show up if hovering over the points and differ for the left and right point. To this end I added a util function `make_hovertext `which overwrites the hovertext set by `layers2traces`.

- I also added some tests and tried to add a visual test (which however failed when running the test while vdiffr::manage_cases showed me a successful doppelganger). Therefore the visual test is commented out in the test.

cpsievert commented on R/layers2traces.R:
```diff
@@ -608,6 +608,65 @@ to_basic.GeomQuantile <- function(data, prestats_data, layout, params, p, ...){
+  # Adds support for ggalt::geom_dumbbell
+  # Implementation follows ggalt::GeomDumbbell
+  # Setup data for left points
+  points.x <- data
```
Please use `_` instead of `.` for separating names

cpsievert commented on R/layers2traces.R:
```diff
@@ -608,6 +608,65 @@ to_basic.GeomQuantile <- function(data, prestats_data, layout, params, p, ...){
+          cbind(x = xend, y = yend, others))
+  })
+  # Set classes
+  if (is.null(params$dot_guide) | !params$dot_guide) {
+    list(
+      prefix_class(data, "GeomPath"),
+      prefix_class(points.x, "GeomPoint"),
+      prefix_class(points.xend, "GeomPoint")
+    )
+  } else {
+    list(
+      prefix_class(dot_df, "GeomPath"),
+      prefix_class(data, "GeomPath"),
+      prefix_class(points.x, "GeomPoint"),
+      prefix_class(points.xend, "GeomPoint")
+    )
+  }
```
```suggestion
  list(
    if (is.null(params$dot_guide) || !params$dot_guide) prefix_class(dot_df, "GeomPath"),
    prefix_class(data, "GeomPath"),
    prefix_class(points.x, "GeomPoint"),
    prefix_class(points.xend, "GeomPoint")
  )
```

cpsievert commented on R/layers2traces.R:
```diff
@@ -1125,3 +1184,41 @@ lty2dash <- function(x) {
+  )
+  data <- list(data)
+  # Copy from layers2traces
+  data <- Map(function(x, y) {
+    if (nrow(x) == 0) return(x)
+    # make sure the relevant aes exists in the data
+    for (i in seq_along(y)) {
+      aesName <- names(y)[[i]]
+      if (!aesName %in% names(x)) next
+      # TODO: should we be getting the name from scale_*(name) first?
+      varName <- y[[i]]
+      # "automatically" generated group aes is not informative
+      if (identical("group", unique(varName, aesName))) next
+      # add a line break if hovertext already exists
+      if ("hovertext" %in% names(x)) x$hovertext <- paste0(x$hovertext, br())
+      # text aestheic should be taken verbatim (for custom tooltips)
+      prefix <- if (identical(aesName, "text")) "" else paste0(varName, ": ")
+      # look for the domain, if that's not found, provide the range (useful for identity scales)
+      txt <- x[[paste0(aesName, "_plotlyDomain")]] %||% x[[aesName]]
+      suffix <- tryNULL(format(txt, justify = "none")) %||% ""
+      if (inherits(x, "GeomBar") && identical(aesName, "y")) {
+        suffix <- format(x[["ymax"]] - x[["ymin"]], justify = "none")
+      }
+      x$hovertext <- paste0(x$hovertext, prefix, suffix)
+    }
+    x$hovertext <- x$hovertext %||% ""
+    x
+  }, data, hoverTextAes)
```
Instead of copy/pasting this from the body of `layers2traces()` let's extract this bit into it's own function that we can use it both places

trekonom commented:
Hi Carson. First. Sorry for messing up the commit history. 

- I folllowed your suggestions for simplifying the code, removing the duplicated code and renaming the variables. 
- After commiting I realized that I thereby introduced a bug. The latest commits fix this bug. I also added an addtional test to check that.
- Finally I removed some code duplication by adding a function `segement2path `. 

Best, Stefan

trekonom commented:
I just added the visual tests to the PR. However, running the visual tests against the docker image failed with the message "there is no package called 'ggalt'". If I should guess ... this probably happens because `ggalt` is not included in the docker image.

cpsievert commented:
Ah, yea, let's add `ggalt` to `Suggests` in the `DESCRIPTION` file
